Package UAA WAR using the Spring Boot plugin

joshuatcasey opened the pull request:
Signed-off-by: Andrew Wittrock <awittrock@pivotal.io>

andrewedstrom commented on uaa/src/main/java/org/cloudfoundry/identity/uaa/UaaBootApplication.java:
```diff
@@ -0,0 +1,15 @@
+    public static void main(String... args) {
+        SpringApplication.run(UaaBootApplication.class, args);
+    }
+
```
Rejecting on account of evil whitespace

andrewedstrom reviewed (changes requested):
It looks like this is failing to compile the tests. We should fix that before merging. Otherwise looks fine to me https://wings.pivotal.io/builds/8635595

joshuatcasey committed:
Use spring-boot-gradle-plugin and BOMs to package the UAA

- spring-boot-dependencies (Spring Boot BOM)
- spring-framework-bom
- spring-security-bom

Signed-off-by: Joshua Casey <jcasey@pivotal.io>
Co-authored-by: Joshua Casey <jcasey@pivotal.io>
